src/CSET/operators/read.py

```diff
@@ -219,6 +219,28 @@ def _load_model(
     cubes = iris.load(
         input_files, constraint, callback=_create_callback(is_ensemble=False)
     )
+    # Check whether we have components of the wind identified by STASH,
+    # (so this will apply only to cubes from the UM), but not the
+    # wind speed and calculate it if it is missing. Note that
+    # this will be biased low in general because the components will mostly
+    # be time averages. For simplicity, we do this only if there is just one
+    # cube of a component. A more complicated approach would be to consider
+    # the cell methods, but it may not be warranted.
+    try:
+        if any([cb.attributes["STASH"] == "m01s03i225" for cb in cubes]) and any(
+            [cb.attributes["STASH"] == "m01s03i226" for cb in cubes]
+        ):
+            if len(
+                cubes.extract(iris.AttributeConstraint(STASH="m01s03i225"))
+            ) == 1 and not any(
+                [cb.attributes["STASH"] == "m01s03i227" for cb in cubes]
+            ):
+                _add_wind_speed_um(cubes)
+            # Convert winds in the UM to be relative to true east and true north.
+            _convert_wind_true_dirn_um(cubes)
+    except (KeyError, AttributeError):
+        pass
```

Comment:
```suggestion
    u_constr = iris.AttributeConstraint(STASH="m01s03i225")
    v_constr = iris.AttributeConstraint(STASH="m01s03i226")
    speed_constr = iris.AttributeConstraint(STASH="m01s03i227")
    try:
        if cubes.extract(u_constr) and cubes.extract(v_constr):
            if len(cubes.extract(u_constr)) == 1 and not cubes.extract(speed_constr):
                _add_wind_speed_um(cubes)
            # Convert winds in the UM to be relative to true east and true north.
            _convert_wind_true_dirn_um(cubes)
    except (KeyError, AttributeError):
        pass
```
Should be functionally identical, but is a bit more readable on the conditions. This is reliant on python treating an empty (cube)list as False, and a list with at least one item as True.

Reply:
A further comment while I'm looking at this, do we need the try and except if we are using cube.extract? I can't see any way a KeyError or AttributeError could be raised.